components/builder-api-client/src/builder.rs

```diff
@@ -499,15 +508,18 @@ impl BuilderAPIProvider for BuilderAPIClient {
     ///
     /// * Remote Builder is not available
     fn delete_origin_secret(&self, origin: &str, token: &str, key: &str) -> Result<()> {
+        debug!("Deleting origin secret: {}, {}", origin, key);
+
         let path = format!("depot/origins/{}/secret/{}", origin, key);
 
-        let res = self.add_authz(self.0.delete(&path), token).send()?;
+        let resp = self.add_authz(self.0.delete(&path), token).send()?;
+        debug!("Status: {:?}", resp.status());
 
-        // We expect NoContent, because the origin must be empty to delete
-        if res.status == StatusCode::NoContent {
-            Ok(())
+        // We expect NO_CONTENT, because the origin must be empty to delete
+        if resp.status() != StatusCode::NO_CONTENT {
+            Err(err_from_response(resp))
         } else {
-            Err(err_from_response(res))
+            Ok(())
```

Comment:
Slight nitpick, but I think the logic for this read better how it originally was:

`if status == NoContent` { Ok } else { Err }`

Reply:
Addressed by other fixes